yt_dlp/compat/__init__.py

```diff
@@ -83,12 +83,54 @@ def compat_expanduser(path):
 WINDOWS_VT_MODE = False if compat_os_name == 'nt' else None
 
 
-def windows_enable_vt_mode():  # TODO: Do this the proper way https://bugs.python.org/issue30075
+def set_windows_conout_mode(new_mode, mask=0xffffffff):
+    # based on https://github.com/python/cpython/issues/74261#issuecomment-1093745755
+    from ctypes import wintypes
+
+    INVALID_HANDLE_VALUE = 0xffffffff  # ((DWORD)-1)
+    STD_OUTPUT_HANDLE = 0xfffffff5  # ((DWORD)-11)
+
+    kernel32 = ctypes.WinDLL('kernel32', use_last_error=True)
+
+    def check_bool(result, func, args):
+        if not result:
+            raise ctypes.WinError(ctypes.get_last_error())
+        return args
+
+    def check_handle(result, func, args):
+        if not result or result == INVALID_HANDLE_VALUE:
+            raise ctypes.WinError(ctypes.get_last_error(), 'Couldn\'t find handle')
+        return args
+
+    kernel32.GetStdHandle.errcheck = check_handle
+    kernel32.GetStdHandle.argtypes = (wintypes.DWORD,)
+    kernel32.GetConsoleMode.errcheck = check_bool
+    kernel32.GetConsoleMode.argtypes = (wintypes.HANDLE, ctypes.POINTER(wintypes.DWORD))
+    kernel32.SetConsoleMode.errcheck = check_bool
+    kernel32.SetConsoleMode.argtypes = (wintypes.HANDLE, wintypes.DWORD)
+
+    hout = kernel32.GetStdHandle(STD_OUTPUT_HANDLE)
+    out = wintypes.DWORD()
+    kernel32.GetConsoleMode(hout, ctypes.byref(out))
+    old_mode = out.value
+    mode = (new_mode & mask) | (old_mode & ~mask)
+    kernel32.SetConsoleMode(hout, mode)
+    return old_mode
+
+
+def windows_enable_vt_mode():
     if compat_os_name != 'nt':
         return
     global WINDOWS_VT_MODE
-    startupinfo = subprocess.STARTUPINFO()
-    startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
-    with contextlib.suppress(Exception):
-        subprocess.Popen('', shell=True, startupinfo=startupinfo).wait()
+    ERROR_INVALID_PARAMETER = 0x0057
+    ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x0004
+    DISABLE_NEWLINE_AUTO_RETURN = 0x0008
+
+    mode = mask = ENABLE_VIRTUAL_TERMINAL_PROCESSING | DISABLE_NEWLINE_AUTO_RETURN
+    try:
+        mode = set_windows_conout_mode(mode, mask)
```

Comment:
Both device handle `19` and `27` fails (due to `ERROR_INVALID_PARAMETER`) while trying to set either `DISABLE_NEWLINE_AUTO_RETURN` or `ENABLE_VIRTUAL_TERMINAL_PROCESSING | DISABLE_NEWLINE_AUTO_RETURN`.